rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -265,10 +266,20 @@ size_t get_placed_texture_storage_size(const rsx::texture &texture, size_t rowPi
 
 std::vector<MipmapLevelInfo> upload_placed_texture(const rsx::texture &texture, size_t rowPitchAlignement, void* textureData)
 {
-	size_t w = texture.width(), h = texture.height();
-	size_t depth = texture.depth();
-	if (depth == 0) depth = 1;
-	if (texture.cubemap()) depth *= 6;
+	u16 w = texture.width(), h = texture.height();
+	u16 depth;
+	u8 layer;
+
+	if (texture.dimension() == 2)
+	{
+		depth = 1;
+		layer = texture.cubemap() ? 6 : 1;
+	}
+	if (texture.dimension() == 3)
```

Comment:
what's if it's not 2 or 3 (could be `else if` here)? Where are depth and layer set?